### PR TITLE
fix(recurring buy): reconfigures how the frequency selection screen works by adding a new endpoint to check if payment method works for a particular frequency

### PR DESCRIPTION
**Pull request by blockdylanb**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/components/Flyout/FrequencyScreen.tsx [hunk 1]
```diff
@@ -2,25 +2,72 @@ import React, { useCallback } from 'react'
...
 import Container from './Container'
 import Content from './Content'
 import Header, { Props as HeaderProps } from './Header'
 import { getPeriodSubTitleText, getPeriodTitleText } from './model'
 
-const FrequencyScreen = ({ children, headerAction, headerMode, setPeriod }: Props) => {
-  // ONE_TIME is not a recurring buy option so take it out before displaying
-  const periods = Object.values(RecurringBuyPeriods)
+const RowDisplay = ({ method, paymentInfo, period, setPeriod }: RowDisplayProps) => {
+  // ONE_TIME is a special case that we need to insert manually
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/components/Flyout/FrequencyScreen.tsx:18]: can you expand on this comment for future us? when is ONE_TIME used and when is it not?

File: packages/blockchain-wallet-v4-frontend/src/modals/RecurringBuys/Frequency/index.tsx [hunk 2]
```diff
@@ -11,27 +12,51 @@ import { RootState } from 'data/rootReducer'
...
+            <FormattedMessage
+              id='modals.recurringbuys.get_started.buy_amount_of_currency'
+              defaultMessage='Buy {amount} of {currency}'
+              values={{ amount, currency }}
+            />
+          </FrequencyScreen>
+        ) : (
+          <></>
+        )}
+      </>
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/modals/RecurringBuys/Frequency/index.tsx:53]: do we need to return fragments on all of these?  would `null` work?

File: packages/blockchain-wallet-v4-frontend/src/modals/RecurringBuys/Frequency/index.tsx [hunk 3]
```diff
@@ -11,27 +12,51 @@ import { RootState } from 'data/rootReducer'
...
-    <FrequencyScreen headerAction={close} headerMode='back' setPeriod={setPeriod}>
-      <FormattedMessage
-        id='modals.recurringbuys.get_started.buy_amount_of_currency'
-        defaultMessage='Buy {amount} of {currency}'
-        values={{ amount, currency }}
-      />
-    </FrequencyScreen>
-  )
+  return data.cata({
+    Failure: (error) => <>{error}</>,
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/modals/RecurringBuys/Frequency/index.tsx:31]: should we style errors better?

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/Frequency/index.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import { actions } from 'data'
+import { RootState } from 'data/rootReducer'
+import { RecurringBuyPeriods } from 'data/types'
+
+import getData from './selectors'
+
+class Frequency extends PureComponent<Props> {
+  constructor(props) {
+    super(props)
+    this.handleFrequencySelection = this.handleFrequencySelection.bind(this)
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/Frequency/index.tsx:16]: should be able to delete this constructor.  you shouldnt need to bind `this` when you define the function with a fat arrow

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/OrderSummary/index.tsx [hunk 5]
```diff
@@ -43,8 +44,8 @@ class OrderSummary extends PureComponent<Props> {
     // first time buyers have 1 tx at this point and RB is set to one time buy so send them to RB walkthrough flow
     if (
       this.props.isRecurringBuy &&
-      this.props.orders.length <= 1 &&
-      this.props.order.period === RecurringBuyPeriods.ONE_TIME &&
+      // this.props.orders.length <= 1 &&
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/OrderSummary/index.tsx:47]: remove?

**Review by schnogz — CHANGES_REQUESTED** [review]

**Review by feccruz — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/components/Flyout/FrequencyScreen.tsx (hunk 1 above)
Comment (feccruz) [reply packages/blockchain-wallet-v4-frontend/src/components/Flyout/FrequencyScreen.tsx:18]: Gostei

**Review by feccruz — APPROVED** [review]

**Comment by feccruz** [comment]
Estou aprendendo cada dia um pouco dessas linguagens aplicados as programações dos internautas.

**Pull request merged**